src/main/java/org/mockito/Mockito.java

```diff
@@ -1278,6 +1278,78 @@
      * </code></pre>
      */
     public static final Answer<Object> CALLS_REAL_METHODS = Answers.CALLS_REAL_METHODS;
+
+    /**
+     * Optional <code>Answer</code> to be used with {@link Mockito#mock(Class, Answer)}.
+     *
+     * Allows Builder mocks to return itself whenever a method is invoked that returns a Type equal
+     * to the class or a superclass.
+     *
+     * <p><b>Keep in mind this answer uses the return type of a method.
+     * If this type is assignable to the class of the mock, it will return the mock.
+     * Therefore if you have a method returning a superclass (for example {@code Object}) it will match and return the mock.</b></p>
+     *
+     * Consider a HttpBuilder used in a HttpRequesterWithHeaders.
+     *
+     * <pre class="code"><code class="java">
+     * public class HttpRequesterWithHeaders {
+
+     *      private HttpBuilder builder;
+
+            public HttpRequesterWithHeaders(HttpBuilder builder) {
+                this.builder = builder;
+            }
+
+            public String request(String uri) {
+                return builder.withUrl(uri)
+                        .withHeader("Content-type: application/json")
+                        .withHeader("Authorization: Bearer")
+                        .request();
+            }
+        }
+
+        private static class HttpBuilder {
+
+            private String uri;
+            private List&lt;String&gt; headers;
+
+            public HttpBuilder() {
+                this.headers = new ArrayList&lt;String&gt;();
+            }
+
+             public HttpBuilder withUrl(String uri) {
+                 this.uri = uri;
+                 return this;
+             }
+
+             public HttpBuilder withHeader(String header) {
+                 this.headers.add(header);
+                 return this;
+             }
+
+             public String request() {
+                return uri + headers.toString();
+             }
+        }
+     * </code></pre>
```

Comment:
My IDE showed the Javadoc without any problems. After quick googling, I stumbled upon http://www.triemax.com/products/jalopy/manual/javadoc.html#d0e19685 which mentions the stars are ignored in `<pre>` blocks. I therefore assume it is no problem. It also makes modifying of code examples easier in Javadoc.